src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -0,0 +1,1425 @@
+        mask = mask.unsqueeze(1)  # expand mask
+        return mask
+
+
+class OwlViTTextModel(OwlViTPreTrainedModel):
+    config_class = OwlViTTextConfig
+
+    def __init__(self, config: OwlViTTextConfig):
+        super().__init__(config)
+        self.text_model = OwlViTTextTransformer(config)
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self) -> nn.Module:
+        return self.text_model.embeddings.token_embedding
+
+    def set_input_embeddings(self, value):
+        self.text_model.embeddings.token_embedding = value
+
+    @add_start_docstrings_to_model_forward(OWLVIT_TEXT_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=OwlViTTextConfig)
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple[Tuple], Tuple[BaseModelOutputWithPooling]]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import OwlViTProcessor, OwlViTTextModel
+
+        >>> model = OwlViTTextModel.from_pretrained("adirik/owlvit-base-patch32")
+        >>> processor = OwlViTProcessor.from_pretrained("adirik/owlvit-base-patch32")
+
+        >>> inputs = processor(text=[["a photo of a cat", "a photo of a dog"], ["photo of a astranout"]], return_tensors="pt")
+        >>> outputs = model(**inputs)
+
+        >>> for output in outputs:  # loop over sets of text queries
+        >>>     last_hidden_state = output.last_hidden_state
+        >>>     pooled_output = output.pooler_output  # pooled (EOS token) states
+        ```"""
+        batch_size = input_ids.shape[0]
+
+        # Get embeddings for all text queries in all batch samples
+        if attention_mask is not None:
+            output = tuple(
+                [
+                    self.text_model(
+                        input_ids=input_ids[idx],
+                        attention_mask=attention_mask[idx],
+                        output_attentions=output_attentions,
+                        output_hidden_states=output_hidden_states,
+                        return_dict=return_dict,
+                    )
+                    for idx in range(batch_size)
+                ]
+            )
+        else:
+            output = tuple(
+                [
+                    self.text_model(
+                        input_ids=input_ids[idx],
+                        attention_mask=None,
+                        output_attentions=output_attentions,
+                        output_hidden_states=output_hidden_states,
+                        return_dict=return_dict,
+                    )
+                    for idx in range(batch_size)
+                ]
+            )
```

Comment:
```suggestion
        # Get embeddings for all text queries in all batch samples
            output = tuple(
                [
                    self.text_model(
                        input_ids=input_ids[idx],
                        attention_mask=attention_mask[idx] if attention_mask is not None else None,
                        output_attentions=output_attentions,
                        output_hidden_states=output_hidden_states,
                        return_dict=return_dict,
                    )
                    for idx in range(batch_size)
                ]
            )
```
Maybe this is a bit simpler